newIDE/app/src/EventsSheet/EventsTree/Renderers/ForEachChildVariableEvent.js

```diff
@@ -0,0 +1,341 @@
+const gd: libGDevelop = global.gd;
+
+const styles = {
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  instructionsContainer: {
+    display: 'flex',
+  },
+  actionsList: {
+    flex: 1,
+  },
+  variableContainer: {
+    marginLeft: '3px',
+    marginRight: '2px',
+  },
+};
+
+type State = {|
+  editingValueIteratorVariableName: boolean,
+  editingKeyIteratorVariableName: boolean,
+  editingIterableVariableName: boolean,
+  anchorEl: ?HTMLSpanElement,
+|};
+
+export default class ForEachChildVariableEvent extends React.Component<
+  EventRendererProps,
+  State
+> {
+  _valueIteratorField: ?SceneVariableField = null;
+  _keyIteratorField: ?SceneVariableField = null;
+  _iterableField: ?SceneVariableField = null;
+
+  state = {
+    editingValueIteratorVariableName: false,
+    editingKeyIteratorVariableName: false,
+    editingIterableVariableName: false,
+    anchorEl: null,
+  };
+
+  edit = (
+    variable: 'iterable' | 'iteratorValue' | 'iteratorKey',
+    anchorEl: HTMLSpanElement
+  ) => {
+    // We should not need to use a timeout, but
+    // if we don't do this, the InlinePopover's clickaway listener
+    // is immediately picking up the event and closing.
+    // Search the rest of the codebase for inlinepopover-event-hack
+    setTimeout(
+      () =>
+        this.setState(
+          {
+            editingValueIteratorVariableName: variable === 'iteratorValue',
+            editingKeyIteratorVariableName: variable === 'iteratorKey',
+            editingIterableVariableName: variable === 'iterable',
+            anchorEl,
+          },
+          () => {
+            // Give a bit of time for the popover to mount itself
+            setTimeout(() => {
+              const field =
+                variable === 'iterable'
+                  ? this._iterableField
+                  : variable === 'iteratorValue'
+                  ? this._valueIteratorField
+                  : this._keyIteratorField;
+              if (field) field.focus();
+            }, 10);
+          }
+        ),
+      10
+    );
+  };
+
+  endEditing = () => {
+    const { anchorEl } = this.state;
+    // Put back the focus after closing the inline popover.
+    // $FlowFixMe
+    if (anchorEl) anchorEl.focus();
+
+    this.setState({
+      editingKeyIteratorVariableName: false,
+      editingValueIteratorVariableName: false,
+      editingIterableVariableName: false,
+      anchorEl: null,
+    });
+  };
+
+  render() {
+    const forEachChildVariableEvent = gd.asForEachChildVariableEvent(
+      this.props.event
+    );
+    const valueIteratorName = forEachChildVariableEvent.getValueIteratorVariableName();
+    const keyIteratorName = forEachChildVariableEvent.getKeyIteratorVariableName();
+    const iterableName = forEachChildVariableEvent.getIterableVariableName();
+
+    return (
+      <div
+        style={styles.container}
+        className={classNames({
+          [executableEventContainer]: true,
```

Comment:
This seems to be missing `largeSelectableArea`/`largeSelectedArea`

Reply:
Indeed, I removed it because it overlapped with the field while I experimented with the styles. I'm adding it back now.